source/common/http/http2/codec_impl.cc

```diff
@@ -344,6 +349,20 @@ void ConnectionImpl::StreamImpl::saveHeader(HeaderString&& name, HeaderString&&
 }
 
 void ConnectionImpl::StreamImpl::submitTrailers(const HeaderMap& trailers) {
+  // Some browsers (e.g. WebKit-based browsers: https://bugs.webkit.org/show_bug.cgi?id=210108) have
+  // a problem with processing empty trailers (END_STREAM | END_HEADERS with zero length HEADERS) of
+  // an HTTP/2 response as reported here: https://github.com/envoyproxy/envoy/issues/10514.
+  const bool skip_encoding_empty_trailers =
+      Runtime::runtimeFeatureEnabled("envoy.reloadable_features.skip_encoding_empty_trailers");
+  if (trailers.empty() && skip_encoding_empty_trailers) {
```

Comment:
So I haven't implemented @lizan's suggestions on adding a flag to `ResponseTrailerMap` which tells whether the trailers map needs to be submitted or not. Hence it is up to filter to decide, even when the map is not empty. If this is an agreeable action that needs to be done I'll do that. cc. @alyssawilk @mattklein123.

Reply:
My personal opinion is that replacing empty trailers with a empty data frame with end_stream true is very likely safe in all cases. I would vote to just do it universally behind a runtime flag and see if anyone complains?